Documentation/policy/language.rst

```diff
@@ -414,6 +419,75 @@ but not CIDR prefix ``10.96.0.0/12``
 
         .. literalinclude:: ../../examples/policies/l3/cidr/cidr.json
 
+.. _DNS based:
+
+DNS based
+---------
+
+``toFQDNs`` simplifies specifying egress policy to IPs of remote, external,
+peers. The DNS lookup for each ``matchName`` is done periodically by
+``cilium-agent`` and the result is used to regenerate endpoint policy. This
+allows tracking changing IPs or sets of IPs that may not be known a priori.
+Despite the naming the ``matchName`` field does not have to be a
+fully-qualified domain name. In cases where the resolver is configured with
+search domains, the DNS lookups from ``cilium`` will not be qualified and will
+utilize the search list.
+
+The DNS lookups are repeated with an interval of 5 seconds, and are made for
+A(IPv4) and AAAA(IPv6) addresses. Should a lookup fail, the most recent IP data
+is used instead. An IP change will trigger a regeneration of the ``cilium``
+policy for each endpoint, and the updated IPs can be seen in the response from
+``cilium policy get``. Each update will also increment the per ``cilium-agent``
+policy repository revision.
+
+``toFQDNs`` rules cannot contain any other L3 rules and can support L4/L7
+rules. 
+
+.. note:: ``toFQDNs`` rules are marked on import with a
+          ``cilium-generated:ToFQDN-UUID`` label. This is for internal
+          bookkeeping and can be safely ignored.
+
+
+.. note:: The DNS resolver must be explicitly whitelisted to allow cilium-agent
+          to send the DNS polls. This is illustrated in the example below.
+
+Example
+~~~~~~~
+
+.. only:: html
+
+   .. tabs::
+     .. group-tab:: k8s YAML
+
+        .. literalinclude:: ../../examples/policies/l3/fqdn/fqdn.yaml
+     .. group-tab:: JSON
+
+        .. literalinclude:: ../../examples/policies/l3/fqdn/fqdn.json
+
+.. only:: epub or latex
+
+        .. literalinclude:: ../../examples/policies/l3/fqdn/fqdn.json
+
+Limitations
+~~~~~~~~~~~
+
+The current ``toFQDNs`` implementation is very limited. It may not behave as expected.
+
+#. The DNS polling is done from the ``cilium-agent process``. This may result
+   in different IPs being returned in the DNS response than those seen by an
+   endpoint or pod.
+
+#. The IP response is used as-is. For DNS responses that return a new IP on
+   every query this may result in a different IP being whitelisted than the one
+   used for current connections.
+
+#. The system configuration determines which DNS resolver is used. When
```

Comment:
Not with the the `clusterWithHostNet` :smile: Can we remove this paragraph then?